### PR TITLE
Feature / Implement engine v1.2

### LINKED ISSUE #75: Show Errors on Board after the PromptForm submit

### LINKED ISSUE #90: Implement the cboard board generator engine

### PR DESCRIPTION
**Pull request by RodriSanchez1**
close #90 and close #75

### CODE REVIEW

File: src/app/[locale]/dashboard/@promptForm/actions.tsx [hunk 1]
```diff
@@ -11,28 +14,78 @@ const promptFormDataSchema = z.object({
...
     return { message: 'Failed to create todo' };
   }
+  if (
+    typeof prompt === 'string' &&
+    typeof rows === 'number' &&
+    typeof columns === 'number'
+  )
+    try {
+      const numberOfTiles = rows * columns;
+      const suggestions =
```
Comment (RodriSanchez1) [inline src/app/[locale]/dashboard/@promptForm/actions.tsx:69]: If the `numberOfTiles `its more dan 15, return an empty array.

File: src/lib/cboard-ai-engine/cboard-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+export const toCboardAdapter = async ({
+  suggestions,
+  columns,
+  rows,
+}: {
+  suggestions: Suggestion[];
+  columns: number;
+  rows: number;
+}): Promise<BoardRecord> => {
+  if (!suggestions.length || !columns || !rows)
```
Comment (RodriSanchez1) [inline src/lib/cboard-ai-engine/cboard-adapter.ts:60]: When `suggestions=[]` enters here and trhows the error. After that i get:

````
 ⨯ Error: Only plain objects, and a few built-ins, can be passed to Client Components from Server Components. Classes or null prototypes are not supported.
    at stringify (<anonymous>)
```

Comment (RodriSanchez1) [reply src/lib/cboard-ai-engine/cboard-adapter.ts:60]: The problem was on the catch block of `onSubmit()`.  I just made a commit to fix it

**Comment by RodriSanchez1** [comment]
@tomivm Take a look! I made some commits. The Pictonizer images are being rendered

File: src/app/[locale]/dashboard/@promptForm/actions.tsx (hunk 1 above)
Comment (tomivm) [reply src/app/[locale]/dashboard/@promptForm/actions.tsx:77]: I open an issue on the Engine repo https://github.com/cboard-org/cboard-ai-engine/issues/22

File: src/lib/cboard-ai-engine/cboard-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import testBoard from '@/dashboard/@board/testBoard.json';
+import { Suggestion, AIImage } from 'cboard-ai-engine';
+
+const generateId = () => Math.floor(Math.random() * 1000000);
+
+const toCboardTilesAdapter = async (
+  tiles: Suggestion[],
+): Promise<TileRecord[]> => {
+  const cboardTiles: TileRecord[] = [];
+  for (const tile of tiles) {
```
Comment (RodriSanchez1) [inline src/lib/cboard-ai-engine/cboard-adapter.ts:13]: Add `await` after for. See [for awiat... of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of)

File: src/lib/cboard-ai-engine/cboard-adapter.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    const id = generateId().toString();
+    const getTileImages = async (pictogram: Suggestion['pictogram']) => {
+      if (pictogram.isAIGenerated) {
+        const images: string[] = [];
+        for (const image of pictogram.images) {
```
Comment (RodriSanchez1) [inline src/lib/cboard-ai-engine/cboard-adapter.ts:18]: Same than above

**Pull request merged**